doc/api/async_hooks.md

```diff
@@ -0,0 +1,677 @@
+# Async Hooks
+
+> Stability: 1 - Experimental
+
+The `async-hooks` module provides an API to register callbacks tracking the
+lifetime of asynchronous resources created inside a Node.js application.
+It can be accessed using `require('async-hooks')`.
+
+## Terminology
+
+An async resource represents either a _handle_ or a _request_.
+
+Handles are a reference to a system resource. Some resources are a simple
+identifier. For example, file system handles are represented by a file
+descriptor. Other handles are represented by libuv as a platform abstracted
+struct, e.g. `uv_tcp_t`. Each handle can be continually reused to access and
+operate on the referenced resource.
+
+Requests are short lived data structures created to accomplish one task. The
+callback for a request should always and only ever fire one time. Which is when
+the assigned task has either completed or encountered an error. Requests are
+used by handles to perform tasks such as accepting a new connection or
+writing data to disk.
+
```

Comment:
It doesn't relate, in fact, it abstracts the two cases into the resource concept.